src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -68,32 +62,106 @@ private TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccoun
+            
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
-            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(original.BuildFromTableAttribute, null, original.CollectAttributeInfo);
 
-            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(original.BuildJObject, null, original.CollectAttributeInfo);
-            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(original.BuildJArray, null, original.CollectAttributeInfo);
+            var bindToExactCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, CloudTable>(
+                original.BindToCloudTable,
+                original.ToParameterDescriptorForCollector,
+                original.CollectAttributeInfo);
+
+            // Includes converter manager, which provides access to IQueryable<ITableEntity>
+            var bindToExactTestCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, IStorageTable>(
+                original.BindToTestCloudTable,
+                original.ToParameterDescriptorForCollector,
+                original.CollectAttributeInfo);
+
+            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(
+                original.BuildFromTableAttribute,
+                null,
+                original.CollectAttributeInfo);
 
-            // Filter to just support JObject, and use legacy bindings for everything else. 
-            // Once we have ITableEntity converters for pocos, we can remove the filter. 
-            // https://github.com/Azure/azure-webjobs-sdk/issues/887
-            bindAsyncCollector = bindingFactory.AddFilter<TableAttribute>(
-                (attr, type) => (type == typeof(IAsyncCollector<JObject>) || type == typeof(ICollector<JObject>)), 
-                bindAsyncCollector); 
+            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(
+                original.BuildJObject,
+                null,
+                original.CollectAttributeInfo);
+
+            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(
+                original.BuildJArray,
+                null,
+                original.CollectAttributeInfo);
 
             var bindingProvider = new GenericCompositeBindingProvider<TableAttribute>(
-                new IBindingProvider[] { bindToJArray, bindToJobject, bindAsyncCollector, original });
+                ValidateAttribute, nameResolver,
+                new IBindingProvider[]
+                {
+                    AllowMultipleRows(bindingFactory, bindAsyncCollector),
+                    AllowMultipleRows(bindingFactory, bindToExactCloudTable),
+                    AllowMultipleRows(bindingFactory, bindToExactTestCloudTable),
+                    bindToJArray,
+                    bindToJobject,
+                    original
```

Comment:
Just single-entity converters. Those are especially tricky because Table has dedicated logic for deciding whether to push any changes back after method returns.